pay/contracts/subscription/HectorSubscription.sol

```diff
@@ -0,0 +1,526 @@
+                isActiveForNow = true;
+                chargeAmount = 0;
+            } else {
+                isActiveForNow = false;
+                chargeAmount = amount - balance;
+            }
+        }
+    }
+
+    /* ======== USER FUNCTIONS ======== */
+
+    function deposit(address _token, uint256 _amount) public nonReentrant {
+        balanceOf[msg.sender][_token] += _amount;
+
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
+
+        emit PayerDeposit(msg.sender, _token, _amount);
+    }
+
+    function createSubscription(uint256 _planId) public onlyValidPlan(_planId) {
+        Subscription storage subscription = subscriptions[msg.sender];
+
+        // check if no or expired subscription
+        if (subscription.planId > 0) {
+            syncSubscription(msg.sender);
+
+            if (subscription.expiredAt > block.timestamp)
+                revert ACTIVE_SUBSCRIPTION();
+        }
+
+        Plan memory plan = plans[_planId];
+
+        // Pay first plan
+        if (balanceOf[msg.sender][plan.token] < plan.amount)
+            revert INSUFFICIENT_FUND();
+
+        unchecked {
+            balanceOf[msg.sender][plan.token] -= plan.amount;
+        }
+
+        IERC20(plan.token).safeTransfer(treasury, plan.amount);
+
+        // Set subscription
+        subscription.planId = _planId;
+        subscription.expiredAt = uint48(block.timestamp) + plan.period;
+
+        emit SubscriptionCreated(msg.sender, _planId, subscription.expiredAt);
+    }
+
+    function depositAndCreateSubscription(uint256 _planId, uint256 _amount)
+        external
+    {
+        deposit(plans[_planId].token, _amount);
+        createSubscription(_planId);
+    }
+
+    function syncSubscriptions(address[] memory froms) external {
+        uint256 length = froms.length;
+        for (uint256 i = 0; i < length; i++) {
+            syncSubscription(froms[i]);
+        }
+    }
+
+    function syncSubscription(address from) public {
+        Subscription storage subscription = subscriptions[from];
+        uint256 planId = subscription.planId;
+
+        // inactive subscription
+        if (planId == 0) return;
+
+        // before expiration
+        if (block.timestamp < subscription.expiredAt) return;
+
+        // after expiration
+        Plan memory plan = plans[planId];
+        uint256 count = (block.timestamp -
+            subscription.expiredAt +
+            plan.period) / plan.period;
+        uint256 amount = plan.amount * count;
+        uint256 balance = balanceOf[from][plan.token];
+
+        // not active for now
+        if (balance < amount) {
+            count = balance / plan.amount;
+            amount = plan.amount * count;
+        }
+
+        if (count > 0) {
+            unchecked {
+                balanceOf[from][plan.token] -= amount;
+                subscription.expiredAt += uint48(plan.period * count);
+            }
+
+            IERC20(plan.token).transfer(treasury, amount);
```

Comment:
@bejaxer ignores return value by IERC20(plan.token).transfer(treasury,amount)
Reason: The return value of an external transfer/transferFrom call is not checked
https://github.com/crytic/slither/wiki/Detector-Documentation#unchecked-transfer